internal/runner/rspec.go

```diff
@@ -89,17 +92,27 @@ func (r Rspec) Run(testCases []string, retry bool) (RunResult, error) {
 		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
 	}
 
-	// Create a temporary file to store the JSON output of the rspec run.
-	// This is a temporary solution, until we have an option to specify the output file.
-	f, err := os.CreateTemp("", "rspec-*.json")
-	if err != nil {
-		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to create temporary file for rspec output: %v", err)
+	var resultPath string
+
+	if r.ResultPath != "" {
+		resultPath = strings.ReplaceAll(r.ResultPath, "*", strconv.FormatInt(time.Now().Unix(), 10))
+	} else {
+		// Create a temporary file if no result path is provided.
+		f, err := os.CreateTemp("", "rspec-*.json")
```

Comment:
I was thinking about "rspec-*.json" could be the default of `BUILDKITE_SPLITTER_RESULT_PATH` and we don't need the conditional check on 97 🤔 the problem is we won't be able to tell if we want to remove the result file at the end. 
Is there value to always keep the result file? or that's more problematic?

Reply:
There is definitely value in keeping the results files around as part of the test splitting runner process. Currently, the `bk/bk` run group outputs N+1 versions of the results JSON files:

One for each run of the tests:

![Screenshot 2024-08-22 at 9 32 46 AM](https://github.com/user-attachments/assets/9aacf90f-9680-45ed-a6f6-f7dfaf3956a1)

And one file used internally by the test splitter to identify which tests need to be retried. 

If we merged this PR, we would have support for persisting the rspec run results _within_ the test splitter, so we could remove the explicit behaviour to store the rspec output results. I think this is currently handled in .rspec.ci.

Anyways TLDR is a little bit simpler and probably valuable for other customers too, so we might as well do it!

Reply:
> Is there value to always keep the result file? or that's more problematic?

I think this is related to @niceking comment

> I think it would possibly be a bit surprising if we created a file and didn't tell the user about it. I think I'd maybe design it like if the customer specifies a file name, then the file is saved, but if they don't, the file is deleted? I wouldn't want a side effect of using the TSC be that we clutter a host machine with lots of files that the user didn't know we were creating? What do you think?


So I think, we should persist the file only when customer specify it?

Reply:
Yep, the comment Rose made actually answered my question. I agree we persist the file only when customer specify it.